### PR TITLE
Directional Firelock Spawning

### PR DESCRIPTION
**Pull request by BurgerLUA**
# Overview
Firelocks now generate their direction based on its surroundings. This only applies to firelocks that spawn in the map. This is entirely a cosmetic feature and any bugs that will (incredibly unlikely) occur will not cause a max singularity to spawn.

https://i.imgur.com/msEMNR2.png

## Issues
Generation does not support T junctions so some corners may look odd, as it picks one of the three possible, with bias towards a specific direction.

### CODE REVIEW

**Comment by BotBOREALIS** [comment]
Icons with diff:
<details><summary>icons/obj/doors/DoorHazard.dmi</summary>

Key | Old | New | Status
--- | --- | --- | ---
alert_cold_1[1]|![alert_cold_1[1]](http://images.aurorastation.org/images/7a70ca69c09d456b236abc8b9f01e76b.png)|![alert_cold_1[1]](http://images.aurorastation.org/images/b464e85362f2c0f6930f397903a85034.png)|Modified
alert_cold_1[2]|![alert_cold_1[2]](http://images.aurorastation.org/images/0a65d9a6b110f59e9bfbc7a0ce4d5d6e.png)|![alert_cold_1[2]](http://images.aurorastation.org/images/01ae4e1b2d5d0dab36dd4ba6acf13bf1.png)|Modified
alert_cold_1[3]|![alert_cold_1[3]](http://images.aurorastation.org/images/fbb7061a4c35e484d063cce56b25298f.png)|![alert_cold_1[3]](http://images.aurorastation.org/images/01ae4e1b2d5d0dab36dd4ba6acf13bf1.png)|Modified
alert_cold_2[1]|![alert_cold_2[1]](http://images.aurorastation.org/images/b464e85362f2c0f6930f397903a85034.png)|![alert_cold_2[1]](http://images.aurorastation.org/images/7a70ca69c09d456b236abc8b9f01e76b.png)|Modified
alert_cold_2[2]|![alert_cold_2[2]](http://images.aurorastation.org/images/bc4c36fa3191bbc27991ae2b9ac74121.png)|![alert_cold_2[2]](http://images.aurorastation.org/images/05fc7e7dbfe77682f67b4ad318d75aff.png)|Modified
alert_cold_2[3]|![alert_cold_2[3]](http://images.aurorastation.org/images/46324d0ed5c7a10ee79e404a48d74fc3.png)|![alert_cold_2[3]](http://images.aurorastation.org/images/05fc7e7dbfe77682f67b4ad318d75aff.png)|Modified
alert_cold_4[1]|![alert_cold_4[1]](http://images.aurorastation.org/images/652c0c193448109fa2c859243d691f66.png)|![alert_cold_4[1]](http://images.aurorastation.org/images/8a894a415b2fe9e2824a67f61cdea627.png)|Modified
alert_cold_4[2]|![alert_cold_4[2]](http://images.aurorastation.org/images/01ae4e1b2d5d0dab36dd4ba6acf13bf1.png)|![alert_cold_4[2]](http://images.aurorastation.org/images/bc4c36fa3191bbc27991ae2b9ac74121.png)|Modified
alert_cold_4[3]|![alert_cold_4[3]](http://images.aurorastation.org/images/05fc7e7dbfe77682f67b4ad318d75aff.png)|![alert_cold_4[3]](http://images.aurorastation.org/images/fbb7061a4c35e484d063cce56b25298f.png)|Modified
alert_cold_8[1]|![alert_cold_8[1]](http://images.aurorastation.org/images/8a894a415b2fe9e2824a67f61cdea627.png)|![alert_cold_8[1]](http://images.aurorastation.org/images/652c0c193448109fa2c859243d691f66.png)|Modified
alert_cold_8[2]|![alert_cold_8[2]](http://images.aurorastation.org/images/05fc7e7dbfe77682f67b4ad318d75aff.png)|![alert_cold_8[2]](http://images.aurorastation.org/images/0a65d9a6b110f59e9bfbc7a0ce4d5d6e.png)|Modified
alert_cold_8[3]|![alert_cold_8[3]](http://images.aurorastation.org/images/01ae4e1b2d5d0dab36dd4ba6acf13bf1.png)|![alert_cold_8[3]](http://images.aurorastation.org/images/46324d0ed5c7a10ee79e404a48d74fc3.png)|Modified
alert_hot_1[1]|![alert_hot_1[1]](http://images.aurorastation.org/images/74743f9948e7c2e55d440bc8e6e00c2c.png)|![alert_hot_1[1]](http://images.aurorastation.org/images/6481ad16ea01292fee28d3df51344c67.png)|Modified
alert_hot_1[2]|![alert_hot_1[2]](http://images.aurorastation.org/images/c1ef479d786eaaab88626fb0f9563211.png)|![alert_hot_1[2]](http://images.aurorastation.org/images/84599ed6c5a3ae29ec7057ede9ce4343.png)|Modified
alert_hot_1[3]|![alert_hot_1[3]](http://images.aurorastation.org/images/41a447218c51681f99bbb6e10b7b815a.png)|![alert_hot_1[3]](http://images.aurorastation.org/images/84599ed6c5a3ae29ec7057ede9ce4343.png)|Modified
alert_hot_2[1]|![alert_hot_2[1]](http://images.aurorastation.org/images/6481ad16ea01292fee28d3df51344c67.png)|![alert_hot_2[1]](http://images.aurorastation.org/images/74743f9948e7c2e55d440bc8e6e00c2c.png)|Modified
alert_hot_2[2]|![alert_hot_2[2]](http://images.aurorastation.org/images/dd2e9f464ec8cc12dd6c2ef04a321508.png)|![alert_hot_2[2]](http://images.aurorastation.org/images/fa8c0f4037fb0bb97092e47776d07c1f.png)|Modified
alert_hot_2[3]|![alert_hot_2[3]](http://images.aurorastation.org/images/73c36b3df95b03a537fb0febe03e35ee.png)|![alert_hot_2[3]](http://images.aurorastation.org/images/fa8c0f4037fb0bb97092e47776d07c1f.png)|Modified
alert_hot_4[1]|![alert_hot_4[1]](http://images.aurorastation.org/images/32895275f47a309c62a43b0b86648975.png)|![alert_hot_4[1]](http://images.aurorastation.org/images/999cb949ce7d694082088bcd79ac73f5.png)|Modified
alert_hot_4[2]|![alert_hot_4[2]](http://images.aurorastation.org/images/84599ed6c5a3ae29ec7057ede9ce4343.png)|![alert_hot_4[2]](http://images.aurorastation.org/images/dd2e9f464ec8cc12dd6c2ef04a321508.png)|Modified
alert_hot_4[3]|![alert_hot_4[3]](http://images.aurorastation.org/images/fa8c0f4037fb0bb97092e47776d07c1f.png)|![alert_hot_4[3]](http://images.aurorastation.org/images/41a447218c51681f99bbb6e10b7b815a.png)|Modified
alert_hot_8[1]|![alert_hot_8[1]](http://images.aurorastation.org/images/999cb949ce7d694082088bcd79ac73f5.png)|![alert_hot_8[1]](http://images.aurorastation.org/images/32895275f47a309c62a43b0b86648975.png)|Modified
alert_hot_8[2]|![alert_hot_8[2]](http://images.aurorastation.org/images/fa8c0f4037fb0bb97092e47776d07c1f.png)|![alert_hot_8[2]](http://images.aurorastation.org/images/c1ef479d786eaaab88626fb0f9563211.png)|Modified
alert_hot_8[3]|![alert_hot_8[3]](http://images.aurorastation.org/images/84599ed6c5a3ae29ec7057ede9ce4343.png)|![alert_hot_8[3]](http://images.aurorastation.org/images/73c36b3df95b03a537fb0febe03e35ee.png)|Modified
alert_cold_1[4]||![alert_cold_1[4]](http://images.aurorastation.org/images/2db84af9f60a65d239085922a6966d2e.png)|Created
alert_cold_1[5]||![alert_cold_1[5]](http://images.aurorastation.org/images/2db84af9f60a65d239085922a6966d2e.png)|Created
alert_cold_1[6]||![alert_cold_1[6]](http://images.aurorastation.org/images/2db84af9f60a65d239085922a6966d2e.png)|Created
alert_cold_1[7]||![alert_cold_1[7]](http://images.aurorastation.org/images/2db84af9f60a65d239085922a6966d2e.png)|Created
</details>

**Comment by BurgerLUA** [comment]
!review

**Comment by BurgerLUA** [comment]
This needs to be in November 2018

**Comment by BurgerLUA** [comment]
Oh, just a disclaimer, I thought I mentioned this earlier in the OP but the map changes are basically removing all directional variants of the firelock as it is handled in this new system.

File: code/game/machinery/doors/firedoor.dm [hunk 1]
```diff
@@ -63,11 +70,52 @@
 	areas_added = list(A)
```
[the remark this answers is not in the document]
Comment (BurgerLUA) [reply code/game/machinery/doors/firedoor.dm]: Fixed, though in a different way I'm not sure if you'll like.

[the remark this answers is not in the document]
Comment (BurgerLUA) [reply code/game/machinery/doors/firedoor.dm]: resolved

**Comment by Arrow768** [comment]
You have a commit there that ais "removed map changes", but you still have map changes in the PR.

**Pull request merged**